lib/classes/Service.test.js

```diff
@@ -210,6 +210,32 @@ describe('Service', () => {
       });
     });
 
+    it('should set provider.stage and provider.region from options', () => {
```

Comment:
hmm I don't think we want to do that ... the `provider`/`default` property were meant to be just "defaults" ... the option overwrides already happens in [`validate.js`](https://github.com/serverless/serverless/blob/master/lib/plugins/aws/lib/validate.js#L12)